platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/AbstractProfileCommand.java

```diff
@@ -0,0 +1,216 @@
+    boolean showDevProfiles = false;
+
+    @Reference
+    private ApplicationService applicationService;
+
+    @Reference
+    private FeaturesService featuresService;
+
+    @Reference
+    private BundleService bundleService;
+
+    protected PrintStream console = System.out;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractProfileCommand.class);
+
+    protected Path developmentProfilePath = Paths.get(System.getProperty("ddf.home"), "etc", "admin", "profiles");
+
+    protected static final String DEV_PROFILE_INSTALL_FEATURES = "install-features";
+    protected static final String DEV_PROFILE_UNINSTALL_FEATURES = "uninstall-features";
+    protected static final String DEV_PROFILE_START_APPS = "start-apps";
+    protected static final String DEV_PROFILE_STOP_BUNDLES = "stop-bundles";
+
+    protected void setApplicationService(ApplicationService applicationService) {
+        this.applicationService = applicationService;
+    }
+    protected void setFeaturesService(FeaturesService featuresService) {
+        this.featuresService = featuresService;
+    }
+    protected void setBundleService(BundleService bundleService) {
+        this.bundleService = bundleService;
+    }
+
+
+    @Override
+    public Object execute() throws Exception {
+
+        if (applicationService == null) {
+            throw new IllegalStateException("ApplicationService not found");
+        }
+        if (featuresService == null) {
+            throw new IllegalStateException("FeaturesService not found");
+        }
+        if (bundleService == null) {
+            throw new IllegalStateException("BundleService not found");
+        }
+        try {
+            doExecute(applicationService, featuresService, bundleService);
+        } catch (ApplicationServiceException ase) {
+            printError("Encountered error while trying to perform command. Check log for more details.");
+            LOGGER.debug("Error while performing command.", ase);
+        }
+        return null;
+    }
+
+    /**
+     * Execute profile interaction
+     * be sent to the console and nothing should be returned.
+     *
+     * @param applicationService
+     * @param featuresService
+     * @param bundleService
+     */
+    protected abstract void doExecute(ApplicationService applicationService,
+                            FeaturesService featuresService,
+                            BundleService bundleService)
+            throws Exception;
+
+
+    /**
+     * Get the {@link Feature} that makes up a profile
+     * @param applicationService {@link ApplicationService}
+     * @param profileName Name of profile to get
+     * @return {@link Feature}
+     */
+    protected Optional<Feature> getProfile(ApplicationService applicationService, String profileName) {
+        return Optional.ofNullable(applicationService.getInstallationProfiles().stream()
+                .filter(i -> i.getName().contains(profileName))
```

Comment:
Same comment as in the previous review ... the fact that we use contains() means we match the first one that contains the string. If more than one matches, we will simply automatically install the first one without prompting if that was the right one. Since we do have a list command to list them, I would suggest we simply do an equals() instead of a contains().